test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SWAEndToEndTests.cs

```diff
@@ -0,0 +1,151 @@
+            string token = _fixture.Host.GenerateAdminJwtToken();
+            request.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", token);
+
+            var response = await _fixture.Host.HttpClient.SendAsync(request);
+
+            // the sync request will fail because we're not running in Antares and can't
+            // communicate with the FE, but it suffices to verify that we attempted to send the request
+            var logs = _fixture.Host.GetScriptHostLogMessages().Where(p => p.Category == typeof(FunctionsSyncManager).FullName).ToArray();
+            var log = logs[0].FormattedMessage;
+            Assert.True(log.StartsWith("Making SyncTriggers request"));
+            Assert.True(log.Contains("HttpTrigger-FunctionAuth"));
+            Assert.True(log.Contains("HttpTrigger-Scenarios"));
+        }
+
+        public class TestFixture : EndToEndTestFixture
+        {
+            private TestScopedEnvironmentVariable _scopedEnvironment;
+
+            public TestFixture() : base(@"TestScripts\CSharp", "csharp", RpcWorkerConstants.DotNetLanguageWorkerName)
+            {
+                var testKeyBytes = TestHelpers.GenerateKeyBytes();
+                var testKey = TestHelpers.GenerateKeyHexString(testKeyBytes);
+
+                var settings = new Dictionary<string, string>()
+                {
+                    { "AzureWebEncryptionKey", testKey },
+                    { EnvironmentSettingNames.WebSiteAuthEncryptionKey, testKey },
+                    { "AzureWebJobsStorage", null },
+                    { EnvironmentSettingNames.AzureWebsiteName, "testsite" }
+                };
+                _scopedEnvironment = new TestScopedEnvironmentVariable(settings);
+            }
+
+            public override void ConfigureScriptHost(IWebJobsBuilder webJobsBuilder)
+            {
+                base.ConfigureScriptHost(webJobsBuilder);
+
+                webJobsBuilder.Services.Configure<ScriptJobHostOptions>(o =>
+                {
+                    o.Functions = new[]
+                    {
+                        "HttpTrigger-Scenarios",
+                        "HttpTrigger-FunctionAuth"
+                    };
+                });
+            }
+
+            public override void ConfigureScriptHost(IServiceCollection services)
+            {
+                base.ConfigureScriptHost(services);
+
+                // replace the base mock FunctionsSyncManager
+                var service = services.FirstOrDefault(d => d.ServiceType == typeof(IFunctionsSyncManager));
+                services.Remove(service);
+                services.AddSingleton<IFunctionsSyncManager, FunctionsSyncManager>();
+            }
+
+            public override void ConfigureScriptHost(IConfigurationBuilder configBuilder)
+            {
+                base.ConfigureScriptHost(configBuilder);
+
+                // wipe existing test settings
+                configBuilder.RemoveTestSettings();
```

Comment:
I think an easier way for you to do this would be to just add a parameter to the EndToEndTestFixture ctor "addTestSettings" that defaults to true. You can override this in your suite ctor above. This value would be passed to the TestFunctionHost that the fixture creates, and the calls to AddTestSettings would be skipped when false. That way your two ConfigureScriptHost/ConfigureWebHost overrides aren't needed, nor is the new RemoveTestSettings method.

Reply:
Agreed; that's a cleaner way of doing things.